nodejs/eks/authenticationMode.ts

```diff
@@ -43,7 +43,13 @@ export function validateAuthenticationMode(args: ClusterOptions) {
 
     if (!supportsConfigMap(args.authenticationMode)) {
         configMapOnlyProperties.forEach((prop) => {
-            if (args[prop]) {
+            const arg: any = args[prop];
+            if (arg) {
+                // Permit empty instanceRoles as a specialCase.
+                if ((arg.length === 0) && prop === "instanceRoles") {
```

Comment:
instanceRoles is of type `pulumi.Input<pulumi.Input<aws.iam.Role>[]>`. I think this will crash if the outer Input is not resolved yet.

Reply:
Great find! Let me rejigger this reinstanting some type discipline.